### PR TITLE
python3Packages.labgrid: fix the build

### PR DESCRIPTION
**Pull request by Emantor**
## Description of changes
Switch to pyproject, remove the no longer required patch and use pythonRelaxDepsHook.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/labgrid/default.nix [hunk 1]
```diff
@@ -32,13 +33,9 @@ buildPythonPackage rec {
...
 
-  patches = [
-    # Pyserial within Nixpkgs already includes the necessary fix, remove the
-    # pyserial version check from labgrid.
-    ./0001-serialdriver-remove-pyserial-version-check.patch
-  ];
+  nativeBuildInputs = [ setuptools-scm pythonRelaxDepsHook ];
 
-  nativeBuildInputs = [ setuptools-scm ];
+  format = "pyproject";
```
Comment (natsukium) [inline pkgs/development/python-modules/labgrid/default.nix:38]:
```suggestion
  pyproject = true;
```
Could you please use `pyproject = true;` instead?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

File: pkgs/development/python-modules/labgrid/default.nix [hunk 2]
```diff
@@ -32,13 +33,9 @@ buildPythonPackage rec {
     sha256 = "sha256-yhlBqqCLOt6liw4iv8itG6E4QfIa7cW76QJqefUM5dw=";
   };
 
-  patches = [
-    # Pyserial within Nixpkgs already includes the necessary fix, remove the
-    # pyserial version check from labgrid.
-    ./0001-serialdriver-remove-pyserial-version-check.patch
-  ];
+  nativeBuildInputs = [ setuptools-scm pythonRelaxDepsHook ];
```
Comment (natsukium) [inline pkgs/development/python-modules/labgrid/default.nix:36]: Which dependency is relaxed or removed?

Comment (natsukium) [inline pkgs/development/python-modules/labgrid/default.nix:36]:
```suggestion
  nativeBuildInputs = [
    setuptools
    setuptools-scm
    wheel
    pythonRelaxDepsHook
  ];
```
Could you please add `setuptools` and `wheel` explicitly?
https://github.com/labgrid-project/labgrid/blob/447f4af921143fa3abcf6785f5564f3c24aee1f9/pyproject.toml#L1-L8

File: pkgs/development/python-modules/labgrid/default.nix [hunk 3]
```diff
@@ -32,13 +33,9 @@ buildPythonPackage rec {
     sha256 = "sha256-yhlBqqCLOt6liw4iv8itG6E4QfIa7cW76QJqefUM5dw=";
   };
 
-  patches = [
-    # Pyserial within Nixpkgs already includes the necessary fix, remove the
-    # pyserial version check from labgrid.
-    ./0001-serialdriver-remove-pyserial-version-check.patch
```
Comment (natsukium) [inline pkgs/development/python-modules/labgrid/default.nix:38]: Please remove this file from a file tree if we don't need it.

**Comment by Emantor** [comment]
Thanks for the review @natsukium! I hope I caught all review comments :+1:

File: pkgs/development/python-modules/labgrid/default.nix [hunk 4]
```diff
@@ -32,13 +35,17 @@ buildPythonPackage rec {
...
+  # pythonRelaxDepsHook is used to relax all dependencies. Can be removed with
+  # the next labgrid release, master already switched to more relaxed
+  # dependencies.
```
Comment (pbsds) [inline pkgs/development/python-modules/labgrid/default.nix:40]: `pythonRelaxDepsHook` currently does nothing, since `pythonRelaxDeps` is not set.

Comment (Emantor) [reply pkgs/development/python-modules/labgrid/default.nix:40]: Indeed, removing it also lets the package build. I looked into the docs and code, but couldn't find out why I no longer need to remove the hard dependencies inside the `pyproject.toml`. If someone knows why please post :smile: Pushed a version without `pythonRelaxDepsHook`.

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 266451` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.labgrid</li>
    <li>python310Packages.labgrid.dist</li>
    <li>python311Packages.labgrid</li>
    <li>python311Packages.labgrid.dist</li>
  </ul>
</details>

**Pull request merged**